fun/management/commands/update_proctoru_cache.py

```diff
@@ -0,0 +1,50 @@
+# -*- coding: utf-8 -*-
+import datetime
+import logging
+import requests
+from optparse import make_option
+
+from django.core.management.base import BaseCommand
+from backoffice import utils_proctorU_api
+
+
+logger = logging.getLogger(__name__)
+
+
+class Command(BaseCommand):
+    help = """Updates the proctoru API cache for the given days
+    """
+    option_list = BaseCommand.option_list + (
+        make_option('-l', '--log',
+                    action='store_true',
+                    dest='log',
+                    default=False),
+        make_option('-d', '--dry-run',
+                    action='store_true',
+                    dest='dry_run',
+                    default=False),
+                    )
+
+    def handle(self, *args, **options):
+        collection = utils_proctorU_api.get_proctoru_api_collection()
+        begin = datetime.datetime.today() - datetime.timedelta(days=200)
+        end = datetime.datetime.today()
+        student_activities = []
+
+        count_init = collection["reports"].count()
+
+        for start, end in utils_proctorU_api.split_large_date_range(begin, end, 30):
+            student_activity = utils_proctorU_api.query_api(requests.post,
+                                         utils_proctorU_api.BASE_URL + utils_proctorU_api.API_URLS["client_activity_report"],
+                                         utils_proctorU_api.request_infos(start, end))
+            if "error" in student_activity:
+                logger.error("Error between {} and {}, nothing was added".format(begin, end))
+                return student_activity
+
+            student_activities.append(student_activity)
+
+        [collection["reports"].insert(sa) for sa in student_activities]
+        import IPython; IPython.embed()
+        logger.info('\n\nAdded {} items to reports\n'.format(collection["reports"].count() - count_init))
```

Comment:
J'ai jamais compris pourquoi on préfixe les messages de logging par '\n' dans les commandes django. Pourquoi ne pas plutôt écrire : 

    logger.info('Added {} items to reports'.format(collection["reports"].count() - count_init))